ArchUnitNET/Loader/LoadTasks/AddMethodDependencies.cs

```diff
@@ -193,6 +196,15 @@ var calledMethodMember in calledMethodMembers
                 yield return new MethodCallDependency(methodMember, calledMethodMember);
             }
 
+            foreach (
+                var genericArgument in genericArgumentTypes
+                    .Where(instance => !instance.Type.IsCompilerGenerated)
+                    .Distinct()
+            )
+            {
+                yield return new MethodSignatureDependency(methodMember, genericArgument);
```

Comment:
`MethodSignatureDependency` is the wrong type here. This dependency type is currently used to describe dependencies of a method to a type through the method's signature. From the existing dependency types, I think `BodyTypeMemberDependency` would fit best.